File: redsheet-portal/src/app/database-models/v-target.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vTarget {
    constructor(vTarget?: vTarget )  {
        let properties = [ "projectNegotiationId", "targetDetail", "targetId",];
        for (let property of properties)
            if (vTarget && vTarget[property])
                this[property] = vTarget[property];
    }
 
	projectNegotiationId : number =   0 ;
 
    @required()
    @maxLength(1000)
	targetDetail : string =   undefined;
 
    @range(1,2147483647)
	targetId : number =   undefined;


}

File: RedSheet.DbEntities/Models/vTarget.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vTargets", Schema = "dbo")]
	[Serializable]
	public partial class vTarget  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.None)] 
	[Key]
	public int ProjectNegotiationId { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string TargetDetail { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int TargetId { get; set; } 

	public vTarget()
	{
    }

    }
}

